File: training-data/python/selection_sort.py
def selection_sort(items):
    items = items[:]
    n = len(items)
    for i in range(n):
        min_idx = i
        for j in range(i + 1, n):
            if items[j] < items[min_idx]:
                min_idx = j
        items[i], items[min_idx] = items[min_idx], items[i]
    return items


if __name__ == "__main__":
    print(selection_sort([5, 3, 8, 1, 9, 2]))

File: training-data/zig/selection_sort.zig
const std = @import("std");

fn selectionSort(items: []i32) void {
    var i: usize = 0;
    while (i < items.len - 1) : (i += 1) {
        var minIdx = i;
        var j: usize = i + 1;
        while (j < items.len) : (j += 1) {
            if (items[j] < items[minIdx]) minIdx = j;
        }
        const tmp = items[i];
        items[i] = items[minIdx];
        items[minIdx] = tmp;
    }
}

pub fn main() void {
    var data = [_]i32{ 5, 3, 8, 1, 9, 2 };
    selectionSort(&data);
    std.debug.print("{any}\n", .{data});
}
